controllers/authController.js

```diff
@@ -19,57 +19,69 @@ const signUp = async (req , res)  =>{
        res.status(422).json({error:"password must not be less than six characters"})
     }
     
-    await  Users.findOne({email : email})
-    .then(user =>{
-      if(user){
+const Email = await Users.findOne({email : email })
+
+    if (Email){
         res.status(422).json({error:"Email already registered"})
-      } else {
+      } 
 
-const newUser = new Users({name , email , username ,  password , password2})
+const User = await Users.findOne({username : username})
+
+    if(User){
+
+        res.status(422).json({error:"Username already exists!"})
 
-        //Hash password
 
-        bcrypt.genSalt(10 , ( err , salt )=>
+    }  else {
+const newUser = new Users({name , email , username ,  password , password2})
+        //Hash password
+        bcrypt.genSalt(10 , ( err , salt)=>
             bcrypt.hash(newUser.password, salt , async ( err , hash) =>{
-                if (err) throw err;
+                if(err) throw err;
                 newUser.password = hash
                 // save 
                 await newUser.save();
                 res.status(200).json("Registered");
             })
         )
```

Comment:
async-await pattern here

Reply:
I have used async -await pattern it wasn't working this is the best

Reply:
It did not work because you did not use it well. 

![image](https://user-images.githubusercontent.com/54318933/179359633-d6e0fd2a-3541-45cd-b1ce-b0c61cee701d.png)


Change it.